packages/aws-rfdk/lib/deadline/lib/version-query.ts

```diff
@@ -139,9 +139,10 @@ export class VersionQuery extends VersionQueryBase {
       resourceType: 'Custom::RFDK_DEADLINE_INSTALLERS',
     });
 
-    this.majorVersion = Token.asNumber(deadlineResource.getAtt('MajorVersion'));
-    this.minorVersion = Token.asNumber(deadlineResource.getAtt('MinorVersion'));
-    this.releaseVersion = Token.asNumber(deadlineResource.getAtt('ReleaseVersion'));
+    const components = props?.version?.split('.').map(x => parseInt(x));
+    this.majorVersion = components ? components[0] : Token.asNumber(deadlineResource.getAtt('MajorVersion'));
```

Comment:
I am verifying not empty value but undefined. 
Maybe such line will be more clear?

Reply:
Hmm, it might be more readable to assign `components` in the undefined case:

```ts
const components = props?.version?.split('.').map(x => parseInt(x)) ?? [];
```

This way `components` is guaranteed to be an array and we can simplify the following statements. For example:

```ts
this.majorVersion = components.length > 0 ? components[0] : Token.asNumber(deadlineResource.getAtt('MajorVersion'));
```